src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -73,7 +73,12 @@ public object GetService(Type serviceType)
             var callCount = 0;
             return provider =>
             {
-                if (Interlocked.Increment(ref callCount) == 2)
+                // PERF: Singleton services once created will always be available in the lookup dictionary.
+                // Hence it is not needed to create an lambda expression for looking up in the dictionary.
+                // This change reduces the memory allocations during the start up because 
+                // lambda expression compilation involves expensive code access security checks for Desktop CLR.
+                var singletonCallSite = callSite as SingletonCallSite;
```

Comment:
The only changes would be for `IServiceProvider.GetService` directly for a singleton. Plus it happens out-of-band (in a Task, not in the main-line code path). Is this a realistic optimization?